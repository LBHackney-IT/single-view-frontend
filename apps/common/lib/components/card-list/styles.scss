@import "~lbh-frontend/lbh/base";
@import "~govuk-frontend/govuk/helpers/media-queries";

.mtfh-card-list {
  display: grid;
  grid-template-columns: repeat(1, minmax(0, 1fr));
  gap: 1rem;
  align-items: flex-start;
  overflow: hidden;
  margin-bottom: 2rem;

  @include govuk-media-query($from: tablet) {
    grid-template-columns: repeat(2, minmax(0, 1fr));
  }

  @include govuk-media-query($from: desktop) {
    grid-template-columns: repeat(2, minmax(0, 1fr));
  }

  @include govuk-media-query($from: 992px) {
    grid-template-columns: repeat(3, minmax(0, 1fr));
  }

  > * {
    margin: 0;
  }
}
