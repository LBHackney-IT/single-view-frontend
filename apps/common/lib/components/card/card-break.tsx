import React from "react";

export const CardBreak = () => {
  return <hr className="mtfh-card-break" />;
};
