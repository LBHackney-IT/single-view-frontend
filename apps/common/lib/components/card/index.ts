export * from "./card-break";
export * from "./card";
export * from "./card-rows";
