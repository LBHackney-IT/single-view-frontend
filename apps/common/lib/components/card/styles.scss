@import "~lbh-frontend/lbh/base";
@import "~govuk-frontend/govuk/helpers/media-queries";

.mtfh-card {
  padding: 10px;
  border: 1px solid rgba(173, 171, 171, 0.4);
  border-radius: 3px;

  > * {
    margin: 6px 0 0;
  }

  dd {
    color: lbh-colour("lbh-grey-1");
  }

  a {
    @include lbh-heading-h4;
    margin: 0;
  }

  &-break {
    border: none;
    border-bottom: 1px solid #bfc1c3;
    margin: 10px 0;
    width: 100%;
    max-width: 160px;
  }
}
