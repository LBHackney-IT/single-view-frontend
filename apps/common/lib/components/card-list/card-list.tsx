import React from "react";
import "./styles.scss";

export const CardList = ({ children }: any): JSX.Element => {
  return <div className="mtfh-card-list">{children}</div>;
};
