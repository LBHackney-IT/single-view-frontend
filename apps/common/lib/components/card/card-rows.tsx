import React from "react";

import { SummaryList, SummaryListItem } from "../summary-list";
import "./styles.scss";

export type CardRow = {
  label: string;
  value: string;
};

export interface CardRowsProp {
  rows: CardRow[];
}

export const CardRows = ({ rows }: CardRowsProp) => {
  return (
    <SummaryList variant="inline">
      {rows.map((row, index) => (
        <SummaryListItem key={index} title={`${row.label}:`}>
          {row.value}
        </SummaryListItem>
      ))}
    </SummaryList>
  );
};
