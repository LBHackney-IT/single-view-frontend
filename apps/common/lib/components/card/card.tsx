import React, { ComponentPropsWithoutRef, forwardRef } from "react";

import cn from "classnames";

import type * as Polymorphic from "@radix-ui/react-polymorphic";

import "./styles.scss";

export type CardComponent = Polymorphic.ForwardRefComponent<
  "div",
  ComponentPropsWithoutRef<"div">
>;

export const Card: CardComponent = forwardRef(function Link(
  { as: CardComp = "div", className, children },
  ref,
) {
  return (
    <CardComp ref={ref} className={cn("mtfh-card", className)}>
      {children}
    </CardComp>
  );
});
