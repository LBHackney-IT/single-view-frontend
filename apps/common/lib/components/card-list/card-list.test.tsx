import React from "react";

import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";

import { Card, CardBreak, CardRows } from "../card";
import { CardList } from "./card-list";

test("CardList renders", () => {
  const rows = [
    { value: "value1", label: "Label 1" },
    { value: "value2", label: "Label 2" },
  ];
  const { container } = render(
    <CardList>
      <Card>
        <CardRows rows={rows} />
        <CardBreak />
        <p>Additional content</p>
      </Card>
    </CardList>,
  );
  expect(screen.getByText("value1")).toBeInTheDocument();
  expect(screen.getByText("Label 1:")).toBeInTheDocument();
  expect(screen.getByText("value2")).toBeInTheDocument();
  expect(screen.getByText("Label 2:")).toBeInTheDocument();
  expect(screen.getByText("Additional content")).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
