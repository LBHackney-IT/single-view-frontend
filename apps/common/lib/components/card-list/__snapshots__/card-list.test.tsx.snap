// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders 1`] = `
<div>
  <div
    class="mtfh-card-list"
  >
    <div
      class="mtfh-card"
    >
      <dl
        class="govuk-summary-list govuk-summary-list--no-border mtfh-summary-list--inline lbh-summary-list"
      >
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Label 1:
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            value1
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Label 2:
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            value2
          </dd>
        </div>
      </dl>
      <hr
        class="mtfh-card-break"
      />
      <p>
        Additional content
      </p>
    </div>
  </div>
</div>
`;
